packages/node/src/login/oidc/oidcHandlers/RefreshTokenOidcHandler.ts

```diff
@@ -56,6 +58,49 @@ function validateOptions(
   );
 }
 
+async function refreshAccess(
+  refreshOptions: RefreshOptions,
+  dpop: boolean
+): Promise<TokenSet & { fetch: typeof fetch }> {
+  let authFetch: typeof fetch;
+  // eslint-disable-next-line camelcase
+  let tokens: TokenSet & { access_token: string };
+  let dpopKey: JWK.ECKey;
+  if (dpop) {
+    dpopKey = await JWK.generate("EC", "P-256");
+    tokens = await refreshOptions.tokenRefresher.refresh(
+      refreshOptions.sessionId,
+      refreshOptions.refreshToken,
+      dpopKey
+    );
+  } else {
+    tokens = await refreshOptions.tokenRefresher.refresh(
+      refreshOptions.sessionId,
+      refreshOptions.refreshToken
+    );
+  }
+  // Rotate the refresh token if applicable
```

Comment:
Could you help me understand what this means? I thought the `.refresh()` call above already fetched a new refresh token if applicable?

Reply:
Rotating the refresh token means that on each use of a refresh token to get an access token, the IdP also invalidates the consumed refresh token, and returns a new one alongside the refreshed access token. 
 
The `.refresh` indeed has a `refresh_token` in the returned `tokens: TokenSet` if the IdP returns one as part of the refresh flow. That means that the original refresh token `refreshOptions.refreshToken` is no longer usable, and this code snippet ensures that the new refresh token `tokens.refresh_token` is given to the `fetch` function so that it can refresh its access token when it expires. If no new refresh token is provided, we can only assume that the previous  `refreshOptions.refreshToken` is still valid.

Reply:
Ah right, so the refresh token passed to us by the developer might be invalid now, in which case we replace it by the new refresh token provided by the IdP. Got it, thanks!

Reply:
Hhmmmm... seems to warrant an explanation in the code itself though, as this is tricky behaviour...?

Reply:
(...in particular I mean the implicit expectation that no new refresh_token from the IdP means we assume the previous one if still 'good').